### PR TITLE
Add Mainmenu including four buttons

### PR DESCRIPTION
**Pull request by abc013**
Based on #12. Tackles parts of #14.

It currently looks like this:
grafik

Contains four buttons:
- play story -> story game
- play endless -> endless game
- settings -> /
- exit game -> exit

---

To continue development here, it would be nice to get #13 merged.

### CODE REVIEW

**Comment by abc013** [comment]
Removed the review label. We can merge but I could implement more functioanlity after #13, as already mentioned.

**Comment by abc013** [comment]
Updated. also increased font size:
grafik

**Comment by Wolkenfarmer** [comment]
I have edited most of the UI (except for the button functionality, which works as expected from abc013 ^^)

Therefore, it should be more appropiate for @abc013 or @midoriusagi to review this PR now.

File: Assets/Scripts/Menu/MenuController.cs [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+	void equalButtonLabelFontSize()
+	{
+		List<float> fontSizes = new List<float>();
+		foreach (GameObject label in buttonLabels)
+		{
+			float fontSize = label.GetComponent<TextMeshProUGUI>().fontSize;
+			fontSizes.Add(fontSize);
+		}
+		float smallestFontSize = fontSizes.Min();
```
Comment (abc013) [inline Assets/Scripts/Menu/MenuController.cs:35]:
```c#
		var smallestFontSize = float.MaxValue;
		foreach (GameObject label in buttonLabels)
		{
			var fontSize = label.GetComponent<TextMeshProUGUI>().fontSize;
			if (fontSize < smallestFontSize)
				smallestFontSize = fontSize;
		}
```

File: Assets/Scripts/Menu/MenuController.cs [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+		foreach (GameObject label in buttonLabels)
+		{
+			label.GetComponent<TextMeshProUGUI>().autoSizeTextContainer = false;
+			label.GetComponent<TextMeshProUGUI>().fontSizeMax = (int)Mathf.Floor(smallestFontSize);
```
Comment (abc013) [inline Assets/Scripts/Menu/MenuController.cs:40]:
```cs
			var mesh = label.GetComponent<TextMeshProUGUI>();
			mesh .autoSizeTextContainer = false;
			mesh .fontSizeMax = (int)Mathf.Floor(smallestFontSize);
```

File: Assets/Scripts/Menu/MenuController.cs [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+
+	void Start()
+	{
+		buttonLabels = new List<GameObject>
+		{
+			StoryLabel,
+			EndlessLabel,
+			SettingsLabel,
+			ExitLabel
+		};
```
Comment (abc013) [inline Assets/Scripts/Menu/MenuController.cs:22]: Why is this list even necessary? As far as I can see, `equalButtonLabelSize` is only called once right below. Why not make 4 calls and send the corresponding label with it?
This way, the for loop below can be removed as well (you can ignore my comments then).

File: UserSettings/EditorUserSettings.asset [hunk 4]
```diff
@@ -9,6 +9,9 @@ EditorUserSettings:
       value: 22424703114646680e0b0227036c6c111b07142f1f2b233e2867083debf42d
       flags: 0
     RecentlyUsedScenePath-1:
+      value: 22424703114646680e0b0227036c58111b12563f22213229
+      flags: 0
+    RecentlyUsedScenePath-2:
```
Comment (abc013) [inline UserSettings/EditorUserSettings.asset:14]: Isn't this supposed to be in `.gitignore`??

**Review by abc013 — COMMENTED** [review]
Code looks good otherwise. Haven't tested ingame.

File: Assets/Scripts/Menu/MenuController.cs (hunk 1 above)
Comment (Wolkenfarmer) [reply Assets/Scripts/Menu/MenuController.cs:35]: Yes, seems a lot better, thanks!
Will be in the next clean-up commit (if I don't manage to amend this to the last commit)

File: Assets/Scripts/Menu/MenuController.cs (hunk 2 above)
Comment (Wolkenfarmer) [reply Assets/Scripts/Menu/MenuController.cs:40]: You perfectionist.. :P
Will be in the next clean-up commit (if I don't manage to amend this to the last commit)

File: UserSettings/EditorUserSettings.asset (hunk 4 above)
Comment (Wolkenfarmer) [reply UserSettings/EditorUserSettings.asset:14]: Good point
I don't know how this managed to get through ^.^'

File: Assets/Scripts/Menu/MenuController.cs (hunk 3 above)
Comment (Wolkenfarmer) [reply Assets/Scripts/Menu/MenuController.cs:22]: It's easier to use for large amounts of labels..
This is here not necessarily the case with only 4 labels, but if we want to use this code-snippet for more purposes, I think this is better.

**Comment by Wolkenfarmer** [comment]
I just merge now, because we are running out of time and these are not essential things you mentioned, sorry

**Pull request merged**